addons/mail/static/src/models/thread/thread.js

```diff
@@ -1143,34 +1209,38 @@ function factory(dependencies) {
             // trusted are:
             // - we have no last message (and then no messages at all)
             // - the message it used to compute is the last message we know
-            if (this.orderedMessages.length === 0) {
+            if (this.orderedNonTransientMessages.length === 0) {
                 return this.serverMessageUnreadCounter;
             }
             // from here serverLastMessageId is not undefined because
-            // orderedMessages contain at least one message.
+            // orderedNonTransientMessages contain at least one message.
             if (!this.lastSeenByCurrentPartnerMessageId) {
-                return this.serverMessageUnreadCounter;
+                return this.serverMessageUnreadCounter +
+                    this.orderedNonTransientMessages.filter(message =>
```

Comment:
Didn't you mean to include all messages here, even transient one? Because if transient are after the last read message they should be counted too.

Reply:
I don't know how in real condition you could have transient messages after an unread messages as transient ones are "asked" by commands, so the thread would have been marked as read as they post the command

Reply:
This reasoning looks correct for now but it is too much flux specific so it is better to not rely on it.